### PR TITLE
[ENGA3-1056]: Added feature flag options in the credit card page to switch between credit card forms

### PR DESCRIPTION
**Pull request by aashishgurung**
#### 1. Objective

Provide an option to switch between embedded secure form and the current form.

Jira Ticket: [#1056](https://opn-ooo.atlassian.net/browse/ENGA3-1056)

#### 2. Description of change

Added a field and an input in the credit card form which can be used to change the form view user wants in the checkout page. In the process, we added the reverted changes related to embedded form.

**Default - Form we currently provide**
Screenshot 2566-03-17 at 09 13 45

**Ehanced form enabled (embedded form)**
Screenshot 2566-03-17 at 09 14 34

**🔧 Environments:**

- WooCommerce: v6.7.0
- WordPress: v6.0.2
- PHP version: 8.1
- Omise WooCommerce: 4.29.0

#### 5. Priority of change

High

### CODE REVIEW

File: convert-to-staging.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?php
```
Comment (ajzkk) [inline convert-to-staging.php:1]: This file do not need to be commit, it is part of docker. 
May be we can add in `.gitignore`

File: assets/javascripts/card-form-customization.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+});
+
+document.getElementById('omise-modal').addEventListener('click', (event) => {
+  if (event.target.id == 'omise-modal') {
+    document.querySelector('.omise-modal').style.display = 'none'
+  }
+});
+
+setDesignFormValues();
+handleColorInputChanges();
```
Comment (ajzkk) [inline assets/javascripts/card-form-customization.js:93]: EOF

File: assets/css/card-form-customization.css [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+}
+
+.omise-modal .body {
+  padding: 10px;
+  padding-top: 10px;
+}
+
+.omise-modal .footer {
+  padding: 10px;
+}
```
Comment (ajzkk) [inline assets/css/card-form-customization.css:153]: EOF

File: includes/admin/views/omise-page-card-form-customization.php [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+<script src="<?php echo Omise::OMISE_JS_LINK ?>"></script>
+<script>
+  window.PUBLIC_KEY = `<?php echo $publicKey ?>`;
+  window.CARD_FORM_THEME = `<?php echo $cardFormTheme;  ?>`;
+  window.DEFAULT_FORM_DESIGN = JSON.parse(`<?php echo json_encode($formDesign) ?>`);
+  window.CARD_BRAND_ICONS = JSON.parse(`<?php echo json_encode($cardIcons) ?>`);
+  window.LOCALE = `<?php echo get_locale(); ?>`;
+</script>
+<script src="<?php echo $assetUrl . '/javascripts/omise-embedded-card.js'; ?>"></script>
+<script src="<?php echo $assetUrl . '/javascripts/card-form-customization.js'; ?>"></script>
```
Comment (ajzkk) [inline includes/admin/views/omise-page-card-form-customization.php:194]: EOF

File: templates/myaccount/my-card.php [hunk 5]
```diff
@@ -36,10 +36,25 @@ class='button delete_card'
...
 </div>
+
+<?php if($viewData['embedded_form_enabled']): ?>
+	<script>
+		window.CARD_FORM_THEME = "<?php echo $viewData['cardFormTheme'] ?>";
+		window.FORM_DESIGN = JSON.parse(`<?php echo json_encode($viewData['formDesign']) ?>`);
+		window.CARD_BRAND_ICONS = JSON.parse(`<?php echo json_encode($viewData['cardIcons']) ?>`);
+		window.LOCALE = `<?php echo get_locale(); ?>`;
+	</script>
+<?php endif; ?>
```
Comment (ajzkk) [inline templates/myaccount/my-card.php:60]: EOF

File: convert-to-staging.php (hunk 1 above)
Comment (aashishgurung) [reply convert-to-staging.php]: Yeah. I already removed it.

File: includes/gateway/class-omise-payment-creditcard.php [hunk 6]
```diff
@@ -81,6 +86,31 @@ function init_form_fields() {
...
+					'description' => __( '<span>Try the new secure form to accept card payments. The new form features additional controls to ensure PCI-DSS compliance.<br /><strong><em>Using this form will be mandatory in a future release</em></strong>.</span>', 'omise' ),
+					'default'     => !self::EMBEDDDED_FORM_ENABLED,
+					'options' => array(
+						!self::EMBEDDDED_FORM_ENABLED => __( 'No', 'omise' ),
+						self::EMBEDDDED_FORM_ENABLED => __( 'Yes', 'omise' )
+					)
+				],
+
+				'card_form_theme' => [
+					'title'       => __( 'Enhanced security form theme', 'omise' ),
```
Comment (ajzkk) [inline includes/gateway/class-omise-payment-creditcard.php:101]: Shall we use the same word? `secure form theme`

**Review by ajzkk — APPROVED** [review]

**Pull request merged**